Cesium3DTilesSelection/include/Cesium3DTilesSelection/TileOcclusionRendererProxy.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include "Library.h"
+#include "Tile.h"
+
+#include <cstdint>
+#include <unordered_map>
+#include <vector>
+
+namespace Cesium3DTilesSelection {
+
+class CESIUM3DTILESSELECTION_API TileOcclusionRendererProxy {
+public:
+  /**
+   * @brief Whether the tile's bounding volume is currently occluded.
+   *
+   * @return The occlusion state.
+   */
+  virtual bool isOccluded() const = 0;
+
+  /**
+   * @brief Whether the tile has valid occlusion info available. If this is
+   * false, the traversal may decide to wait for the occlusion result to become
+   * available in future frames.
+   *
+   * Client implementation note: Do not return false if the occlusion for this
+   * tile will _never_ become available, otherwise the tile may not refine
+   * while waiting for occlusion. In such a case return true here and return
+   * false for isOccluded, so the traversal treats the tile as if it is _known_
+   * to be unoccluded.
+   *
+   * @return Whether this tile has valid occlusion info available.
+   */
+  virtual bool isOcclusionAvailable() const = 0;
+
+protected:
+  friend class TileOcclusionRendererProxyPool;
+  /**
+   * @brief Reset this proxy to target a new tile. If nullptr, this proxy is
+   * back in the pool and will not be used for further occlusion until reset
+   * is called again with an actual tile.
+   *
+   * @param pTile The tile that this proxy represents or nullptr if the proxy
+   * is back in the pool.
+   */
+  virtual void reset(const Tile* pTile) = 0;
+
+private:
+  bool _usedLastFrame = false;
+  TileOcclusionRendererProxy* _pNext = nullptr;
+};
+
+class CESIUM3DTILESSELECTION_API TileOcclusionRendererProxyPool {
+public:
+  virtual ~TileOcclusionRendererProxyPool(){};
```

Comment:
`destroyPool` is called in the Tileset destructor, which is needed since `initPool` is called in the Tileset constructors. It probably makes sense to call it here too out of principle, just applied the change.